scripts/submodule-diff.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+git submodule foreach 'git describe HEAD --tags' > pre.log
+git submodule foreach 'git fetch'
+git submodule foreach 'git describe `git log --branches -1 --pretty=format:"%H"` --tags --abbrev=0' > post.log
```

Comment:
default to `$()` instead of backticks (doesn't matter much here, because no nesting, but it should be the preferred style in any case)

Reply:
My shell foo isn't great, is that style compatible across shells? I like backticks because of readability, and have not run into problems with them, but I don't use the shell much.

On line 5, there is some nesting going on, in any case. Although technically maybe it isn't done by the shell?

Reply:
You cannot nest with backticks only. $() is preferred and POSIX standard: https://stackoverflow.com/a/9406350